### ISSUE TITLE
Virus detected

### ISSUE THREAD

**Issue by DerBasler** [coreybutler/nvm-windows#882]
**Issue:**
2 Virus scanner report a Trojan.Cometer in the latest exe

**How To Reproduce:**
Download https://github.com/coreybutler/nvm-windows/releases/download/1.1.10/nvm-setup.exe
Upload it to https://www.virustotal.com/

**Expected Behavior:**
No virus flaged

**Additional context:**
This might be a false positive but it does not happen in the https://github.com/coreybutler/nvm-windows/releases/download/1.1.9/nvm-setup.exe version so this was strange.

**Comment by 0liver** [comment]
[This](https://www.virustotal.com/gui/file/925a577a97e0fe0ab7d93b295e6d0b690d83a3a5b0e7c1204360fc8712401b6c), indeed, looks suspicious:

![image](https://user-images.githubusercontent.com/164360/202447443-37a95a42-81ed-48e4-bbd5-8aa533e0e6c1.png)

But [this scan](https://opentip.kaspersky.com/925A577A97E0FE0AB7D93B295E6D0B690D83A3A5B0E7C1204360FC8712401B6C/results) says the file is clean.

I've also checked the _exe_ file using ESET Online Scanner, and it did not detect any virus, either.

So, maybe, VirusTotal is overly sensitive

**Comment by coreybutler** [comment]
VirusTotal has always been exceptionally sensitive. Nothing changed in the installer, but my quick and dirty research suggests Trojan.Cometer may be flagged by anything that spawns a hidden window. The installer runs commands to check for existing versions of Node [here](https://github.com/coreybutler/nvm-windows/blob/master/nvm.iss#L143) and [here](https://github.com/coreybutler/nvm-windows/blob/master/nvm.iss#L150). It also creates the first symlink [here](https://github.com/coreybutler/nvm-windows/blob/master/nvm.iss#L313). These operations may be inappropriately being picked up as "hidden windows".

Bottom line: nothing nefarious is happening, and nothing has changed in this file (other than version numbers) in considerable time. I'm not sure if there is anything that can be done about this other than registering an exception with Google/Ikarus.

**Comment by 0liver** [comment]
Thanks for commenting! That's already a lot of information. I wasn't aware that:
> Trojan.Cometer may be flagged by anything that spawns a hidden window

I only chimed to make sure nothing interfered with the packaging process. We're looking forward to using NVM4W at Lombiq really soon!

Kind regards!

**Comment by coreybutler** [comment]
> We're looking forward to using NVM4W at Lombiq really soon!

I'm assuming Lombiq is your company. Companies may be interested in the upcoming [Runtime](https://github.com/coreybutler/nvm-windows/wiki/Runtime) effort I'm working on.

**Comment by DevRCRun** [comment]
We had also noted this. Out of interest, are the other names picked up for the nvm.exe file simply a function of it getting extracted by the installer with different temporary names?

is-SSIDO.tmp 
is-ORQJV.tmp 
is-7OU8O.tmp 
is-90S9T.tmp

https://www.virustotal.com/gui/file/ffd8f0c00cbb466a6c2bf3f82ce883e02ee7e769226cc2dab3feca5493e9945e/details 

The IP address listed again with only one objection to it from Comodo Valkyrie Verdict appears primarily Microsoft. I'm guessing this could be from where the node versions / shasums are fetched? (sorry not had too much of a chance to dig into the code).

Thanks for the reminder about Runtime, I'll fill that survey out now

**Comment by coreybutler** [comment]
@DevRCRun - honestly, I have no idea what those temp files are.

One thing I did notice is this:

<img width="378" alt="image" src="https://user-images.githubusercontent.com/770982/203449472-058703da-c0ea-4778-a2b6-31fc3c96a998.png">

It thinks it was compiled with Go 1.15.x when the latest release was compiled with 1.19.0... so something seems off about those temp files. Maybe these tools are pulling an old binary, or maybe the ML complaining of a virus is too sensitive. 

The only other thing I can think of are the other base files found at https://github.com/coreybutler/nvm-windows/tree/master/bin... but there are only 4 of those and 5 temp files.

**Comment by c33s** [comment]
not sure if we can only blame the hidden windows of the installer here. also [`nvm-noinstall.zip`][1] and [`nvm.exe`][2] trigger a warning.

the issue maybe go itself. see https://go.dev/doc/faq#virus and https://groups.google.com/g/golang-nuts/c/lPwiWYaApSU

[1]: https://www.virustotal.com/old-browsers/file/79974E8D60916910FFEE604434668909A40F66B413D6CBEE6BD88F9E868238AB
[2]: https://www.virustotal.com/old-browsers/file/6EEF8B67AEEFCB060C98B1E4925E5B23E6097546B73F564C5839F8DCB2DFBC3B

**Comment by coreybutler** [comment]
With alternative forms of the binary (i.e. the temp files), it could just be that some of the antivirus tools are picking up the name from a different installer (whose process may be a little different, like renaming files after downloading them). The antivirus could also be picking up filenames prematurely, i.e. performing a scan partially through the installation. 

Older versions of Go are sometimes picked up by outdated antivirus platforms. This was prominent in older versions of NVM4W because it was built with older versions of Go. That was 4-5 years ago. At this point, I pretty much chalk it up to the antivirus being incorrect/outdated.

Most of the antivirus failures stem from machine learning (i.e. not verified by a human), which isn't perfect.

Obviously the goal is for NVM4W to pass all antivirus, but some of the vendors really are really just outdated/inaccurate. I think the evidence of this is in the VirusTotal link @DevRCRun posted... because it _used_ to show a problem. Now it shows 100% compliance.

<img width="1016" alt="image" src="https://user-images.githubusercontent.com/770982/206479675-57fa32a3-60b8-4d12-8204-9703e67048c6.png">

Same goes for the link @0liver posted, which now looks like this:

<img width="989" alt="image" src="https://user-images.githubusercontent.com/770982/206480177-c9327f81-582e-41ae-984f-39707d888a79.png">

**Comment by 0liver** [comment]
I'm happy this is over 😀 Thanks for keeping us in the loop, @coreybutler!
